src/test/java/org/jabref/logic/importer/ParserResultTest.java

```diff
@@ -48,4 +54,80 @@ public void doesNotHaveDuplicateWarnings() {
         parserResult.addWarning("Duplicate Warning");
         assertEquals("Duplicate Warning", parserResult.getErrorMessage());
     }
+
+    @Test
+    public void warningAddedForWhitespaceInCitationKeyImport(@TempDir Path tmpDir) throws IOException {
+        // whitespace after citation key "myArticle "
+        String bibtexEntry = """
+                @article{ myArticle ,
+                  author    = "Author Name",
+                  title     = "Title of the Article",
+                  journal   = "Journal Name",
+                  year      = "2024",
+                  pages     = "1-10",
+                  publisher = "Publisher Name"
+                }
+                """;
+        Path tempFile = tmpDir.resolve("invalidBibTex.bib");
+        Files.write(tempFile, bibtexEntry.getBytes());
+        ParserResult parserResult = OpenDatabase.loadDatabase(tempFile, mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
+        assertEquals("Line 1: Found corrupted citation key  (contains whitespaces).", parserResult.warnings().getFirst());
+    }
+
+    @Test
+    public void warningAddedForMissingCommaInCitationKeyImport(@TempDir Path tmpDir) throws IOException {
+        // Comma replaced by whitespace instead in citation key "myArticle "
+        String bibtexEntry = """
+            @article{myArticle\s
+               author    = "Author Name",
+               title     = "Title of the Article",
+               journal   = "Journal Name",
+               year      = "2024",
+               pages     = "1-10",
+               publisher = "Publisher Name"
+             }
+            """;
+        Path tempFile = tmpDir.resolve("invalidBibTex.bib");
+        Files.write(tempFile, bibtexEntry.getBytes());
+        ParserResult parserResult = OpenDatabase.loadDatabase(tempFile, mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
+        assertEquals("Line 1: Found corrupted citation key  (comma missing).", parserResult.warnings().getFirst());
+    }
+
+    @Test
+    public void warningAddedForCorruptedCitationKeyInImport(@TempDir Path tmpDir) throws IOException {
+        String bibtexEntry = """
+            @article{myArticle
+               author    = "Author Name",
+               title     = "Title of the Article",
+               journal   = "Journal Name",
+               year      = "2024",
+               pages     = "1-10",
+               publisher = "Publisher Name"
+             }
+            """;
+
+        Path tempFile = tmpDir.resolve("invalidBibTex.bib");
+        Files.write(tempFile, bibtexEntry.getBytes());
+        ParserResult parserResult = OpenDatabase.loadDatabase(tempFile, mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor());
+        assertEquals("Line 2: Found corrupted citation key .", parserResult.warnings().getFirst());
```

Comment:
Whitespace before the dot on purpose?

Reply:
This is the %1 parameter for the citation key: 
Line\ %0\:\ Found\ corrupted\ citation\ key\ %1.=Line %0: Found corrupted citation key %1.